comictaggerlib/taggerwindow.py

```diff
@@ -1580,26 +1634,32 @@ def copy_tags(self) -> None:
                 failed_list = []
                 success_count = 0
                 for prog_idx, ca in enumerate(ca_list, 1):
-                    QtCore.QCoreApplication.processEvents()
-                    if prog_dialog.wasCanceled():
-                        break
-
-                    prog_dialog.setValue(prog_idx)
-                    prog_dialog.setLabelText(str(ca.path))
-                    QtCore.QCoreApplication.processEvents()
-
-                    if ca.has_metadata(src_style) and ca.is_writable():
-                        md = ca.read_metadata(src_style)
-
-                        if dest_style == "cbi" and self.config[0].Comic_Book_Lover__apply_transform_on_bulk_operation:
+                    ca_saved = False
+                    for style in dest_styles:
+                        if ca.has_metadata(style):
+                            QtCore.QCoreApplication.processEvents()
+                            if prog_dialog.wasCanceled():
+                                break
+
+                            prog_dialog.setValue(prog_idx)
+                            prog_dialog.setLabelText(str(ca.path))
+                            center_window_on_parent(prog_dialog)
+                            QtCore.QCoreApplication.processEvents()
+
+                        if ca.has_metadata(src_style) and ca.is_writable():
+                            md = ca.read_metadata(src_style)
```

Comment:
This needs to be moved out of the loop, no reason to read the same metadata several times

Reply:
You also need to call continue when this check fails otherwise md may not exist when it tries to copy the tags